File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        // use array hashing
        int[] arr = new int[128];
        // traverse
        for(char c: s.toCharArray()){
            arr[c]++;
        }
        // count
        int count = 0;
        int hasOdd = 0;
        for(int i: arr){
            if(i % 2 == 0) count += i;
            else{
                count += i - 1;
                hasOdd = 1;
            }
        }
        return count+hasOdd;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        arr = [0] * 128
        # traverse string
        for c in s:
            arr[ord(c)] += 1
        # count
        count = 0
        hasOdd = 0
        for i in arr:
            if i % 2 == 0:
                count += i 
            else:
                count += i-1
                hasOdd = 1
        return count + hasOdd
        
        